container-core/src/main/java/com/yahoo/jdisc/http/server/jetty/JettyConnectionLogger.java

```diff
@@ -394,6 +402,14 @@ synchronized ConnectionLogEntry toLogEntry() {
             return builder.build();
         }
 
+        private static String certificateFingerprint(byte[] derEncoded) {
+            try {
+                return HexDump.toHexString(MessageDigest.getInstance("SHA-1").digest(derEncoded));
+            } catch (NoSuchAlgorithmException e) {
+                throw new RuntimeException(e);
```

Comment:
It's probably not good to fail with an exception for an issue like this? Log error and return null instead? Or make the fingerprint be the error message? :)

Reply:
The exception is thrown if the JVM implementation does not provide the SHA-1 digest algorithm. That should never happen.